scanners/ssh-audit/parser/parser.test.js

```diff
@@ -0,0 +1,587 @@
+// SPDX-FileCopyrightText: the secureCodeBox authors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+const fs = require("fs");
+const util = require("util");
+const {
+  validateParser,
+} = require("@securecodebox/parser-sdk-nodejs/parser-utils");
+
+
+const readFile = util.promisify(fs.readFile);
+
+const {parse} = require("./parser");
+
+test("ssh-audit parser parses a result into proper findings for dummy-ssh", async () => {
+    const hosts = JSON.parse(
+        await readFile(__dirname + "/__testFiles__/dummy-ssh.json", {
+          encoding: "utf8",
+        })
+      );
+    const findings = await parse(hosts);
+    await expect(validateParser(findings)).resolves.toBeUndefined();
+    expect(findings).toMatchInlineSnapshot(`
+    [
+      {
+        "attributes": {
+          "compression_algorithms": [
+            "none",
+            "zlib@openssh.com",
+          ],
+          "encryption_algorithms": [
+            "chacha20-poly1305@openssh.com",
+            "aes128-ctr",
+            "aes192-ctr",
+            "aes256-ctr",
+            "aes128-gcm@openssh.com",
+            "aes256-gcm@openssh.com",
+          ],
+          "fingerprints": [
+            {
+              "hash": "eLwgzyjvrpwDbDr+pDbIfUhlNANB4DPH9/0w1vGa87E",
+              "hash_alg": "SHA256",
+              "hostkey": "ssh-ed25519",
+            },
+            {
+              "hash": "c8:65:6b:d1:59:03:56:21:d9:0f:84:83:ce:ac:40:86",
+              "hash_alg": "MD5",
+              "hostkey": "ssh-ed25519",
+            },
+            {
+              "hash": "MbRX/CgQyN6/p8/ZjORurfaJqDhu4VEIWfXo0BnxaCE",
+              "hash_alg": "SHA256",
+              "hostkey": "ssh-rsa",
+            },
+            {
+              "hash": "a5:6f:62:26:81:03:b7:5e:06:48:10:04:79:4b:ac:32",
+              "hash_alg": "MD5",
+              "hostkey": "ssh-rsa",
+            },
+          ],
+          "host": "dummy-ssh.demo-targets.svc",
```

Comment:
Should be `hostname`
Also does the result file format contain the info on which port the ssh server is running on?
So that we'd have this here as well (especially interesting for ssh service not running on the default port)

Reply:
We changed it from `hostname` to `host` because ssh-audit doesn't differentiate between IP-address and hostname. So to avoid having `hostname: 127.0...` or `ip-address: dummy-ssh`, we put them together and named it host. Is there a better alternative?

Reply:
We decided we will name it `hostname` to avoid confusion.

Reply:
Mh ok understandable, at least from a usage / consistency perspective I'd prefer to have both a hostname and a ip_address field and have one of them be null depending on what ssh-audit gives us.

This way you can more easily filter in Kibana for finding with `attributes.hostname` and get all results (which have a hostname) and you wouldn't need to search both in host and hostname. If somebody wants a straightforward field which has the location set they should use the `location` field.

Reply:
okay, that makes sense. We changed it accordingly and now have hostname and ip_address.